src/components/combo_box/combo_box.tsx

```diff
@@ -110,6 +110,10 @@ interface _EuiComboBoxProps<T>
    * When `true` only allows the user to select a single option. Set to `{ asPlainText: true }` to not render input selection as pills
    */
   singleSelection: boolean | EuiComboBoxSingleSelectionShape;
+  /**
+   * Display matching options that start with search value on top of the list
+   */
+  startingWith?: boolean;
```

Comment:
My main comment here is that we are trying not to add more boolean props if possible and I think this is a good case to use an enum instead. Something like:
```suggestion
  /**
   * Display matching options by:
   * `startsWith`: moves items that start with search value to top of the list;
   * `none`: don't change the sort order of initial object
   */
  sortMatchesBy?: 'none' | 'startsWith';
```

And defaults to `none`.